openedx/core/djangoapps/course_roles/models.py

```diff
@@ -0,0 +1,92 @@
+"""
+Models for course roles schema
+"""
+from django.contrib.auth import get_user_model
+from django.db import models
+
+from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
+from organizations.models import Organization
+
+User = get_user_model()
+
+
+class CourseRolesRole(models.Model):
+    """
+    Model for a course roles role.
+
+    A role is a collection of permissions that can be assigned to a user.
+    The Service field defines for which service UI the role is intended, such as CMS or LMS.
+
+    """
+    name = models.CharField(max_length=255)
+    short_description = models.CharField(max_length=255)
+    long_description = models.TextField()
+    service = models.ForeignKey('CourseRolesService', on_delete=models.DO_NOTHING, null=True)
+    permissions = models.ManyToManyField('CourseRolesPermission', through='CourseRolesRolePermissions')
+    users = models.ManyToManyField(User, through='CourseRolesUserRole')
+
+    def __str__(self):
+        return self.name
+
+
+class CourseRolesPermission(models.Model):
+    """
+    Model for a course roles permission.
+
+    A permission represents what a user can do.
```

Comment:
A permission represents a single activity a user is allowed to carry out (maybe?)